torch/_dynamo/variables/nn_module.py

```diff
@@ -303,33 +303,29 @@ def call_method(
         key = self.module_key
         module = tx.output.get_submodule(key)
 
-        if name == "__call__":
-            # TODO(whc)  do we really need this special case?
+        if name == "_call_impl":
```

Comment:
i'm curious what the difference between name == "__call__" and name == "_call_impl" is?

Reply:
This is because of:
https://github.com/pytorch/pytorch/blob/ae55619a2bb73ebcdc80b02a6ccd72275a9ce23e/torch/nn/modules/module.py#L1575

If users explicitly use ```self.layer.__call__(x)```, the name argument of ```call_method``` is ```_call_impl``` .